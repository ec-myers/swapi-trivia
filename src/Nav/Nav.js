import React from 'react';

export default Nave