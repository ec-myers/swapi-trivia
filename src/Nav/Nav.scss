nav {
  border: 1px solid red;
  display: flex;
  align-items: center;
  justify-content: space-between;
  div {
    display: flex;
    flex-direction: column;
  }
  button {
    height: 10%;
    width: 20%;
    &:hover {
      cursor: pointer;
    }
  }
}