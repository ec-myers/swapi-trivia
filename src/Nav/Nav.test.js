import React from 'react'
import {shallow} from 'enzyme'
import Nav from './Nav'

describe('Nav', () => {
  let wrapper, mockLogout;
  beforeEach(() => {
  mockLogout = jest.fn()
    let mockUser = {
      name: 'Greg',
      quote: 'I know',
      rank: 'Jedi Knight'
    }
    wrapper = shallow(<Nav
      user={mockUser}
      logOut={mockLogout}
    />)
  })
  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
  it('should call logOut metho prop when link is clicked', () => {
    wrapper.find('Link').simulate('click');
    expect(mockLogout).toHaveBeenCalled()
  })
})
