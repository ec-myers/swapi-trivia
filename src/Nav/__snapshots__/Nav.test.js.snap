// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav should match the snapshot 1`] = `
<nav>
  <div
    className="user-container"
  >
    <div
      className="inner"
    >
      <span>
        i
      </span>
      <p>
        Name: 
        Greg
      </p>
      <p>
        Quote: 
        I know
      </p>
      <p>
        Rank: 
        Jedi Knight
      </p>
    </div>
  </div>
  <NavLink
    activeStyle={
      Object {
        "color": "coral",
      }
    }
    className="nav-link"
    to="/movies"
  >
    Movies
  </NavLink>
  <NavLink
    activeStyle={
      Object {
        "color": "coral",
      }
    }
    className="nav-link"
    to="/favorites"
  >
    Favorites(
    ) 
  </NavLink>
  <Link
    onClick={[MockFunction]}
    to="/"
  >
    <button>
      Logout
    </button>
  </Link>
</nav>
`;
